src/main/java/com/minecolonies/api/util/EntityUtils.java

```diff
@@ -205,7 +206,7 @@ private static boolean checkValidSpawn(@NotNull final BlockGetter world, final B
         for (int dy = 0; dy < height; dy++)
         {
             final BlockState state = world.getBlockState(pos.above(dy));
-            if (!state.is(ModTags.validSpawn) && BlockUtils.isAnySolid(state))
+            if (!state.is(ModTags.validSpawn) && (PathfindingUtils.isLiquid(state) || ShapeUtil.hasCollision(world, pos.above(dy), state)))
```

Comment:
isnt that an || in the beginning too

Reply:
nah because this check is specifically for allowing blocks which are not known as valid and if they are not valid tagged and have liquid or collisions they're excluded